test/close-byo-router.js

```diff
@@ -0,0 +1,48 @@
+const assert = require('assert')
+const childProcess = require('child_process')
+const path = require('path')
+
+describe('Testing jsonapi-server with bring-your-own router', () => {
+  const serverPath = path.join(__dirname, 'fixtures', 'server-byo-router.js')
+
+  let child
+  let isErrorEmitted
+  let isExitEmitted
+  let exitCode
+  let exitSignal
+
+  it('"exit" event, no "error" event', (done) => {
+    setTimeout(() => {
+      assert.equal(isExitEmitted, true)
+      assert.equal(isErrorEmitted, false)
+      done()
+    }, 1000)
+  })
+
+  it('exit code is 0 (success)', () => {
+    assert.equal(exitCode, 0)
+    assert.equal(exitSignal, null)
+  })
+
+  before(() => {
+    child = null
+    isErrorEmitted = false
+    isExitEmitted = false
+    exitCode = null
+
+    child = childProcess.fork(serverPath, [], { stdio: 'inherit' })
+    child.on('error', () => { isErrorEmitted = true })
+    child.on('exit', (code, signal) => {
+      exitCode = code
```

Comment:
I hit a timing issue on Travis when cutting a new release, I ended up making the `before` block async and letting the test progress once the child has exited:
https://github.com/holidayextras/jsonapi-server/pull/226/commits/9e5111ec369427d86b3e3086d867af3ff22f0477
example build failure (only in Node v4, only some of the time):
https://travis-ci.org/holidayextras/jsonapi-server/builds/182226747